gestalt/__init__.py

```diff
@@ -611,11 +596,60 @@ def fetch_vault_secrets(self) -> None:
             try:
                 secret_token = self.vault_client.secrets.kv.v2.read_secret_version(
                     mount_point=str(mount_path), path=secret_path)
+                print(secret_token)
                 self.__conf_data.update(secret_token['data']['data'])
+                if secret_token['lease_id'] != '': 
+                    secret_lease = (secret_token['lease_id'],
+                                secret_token['lease_duration'], time.time())
+                    self.secret_ttl_identifier.append(secret_lease)
+                
             except hvac.exceptions.InvalidPath as err:
                 raise RuntimeError(
                     "Gestalt Error: The secret path or mount is set incorrectly"
                 )
             except requests.exceptions.ConnectionError as err:
                 raise RuntimeError(
                     "Gestalt Error: Gestalt couldn't connect to Vault")
+            except Exception as err:
+                raise RuntimeError(f"Gestalt Error: {err}")
+
+    def generate_database_dynamic_secret(self, mount_point: str, db_name: str, role_name: str):
+        """Generates a dynamic secret for a database and updates the configuration data structure
+
+        Args:
+            mount_point (str): mount_point of the secret engine in vault
+            connection_name (str): database connection name
+            role_name (str): role under which the secret needs to be generated
+        """
+        secret_engines_list = self.vault_client.sys.list_mounted_secrets_engines(
+    )['data'].keys()
+        if f"{mount_point}/" not in secret_engines_list:
+            raise RuntimeError(f"No secrets engine exists at the mount point {mount_point}")
+        existing_connections = self.vault_client.secrets.database.list_connections(mount_point=mount_point)["data"]["keys"]
+        if db_name not in existing_connections:
+            raise RuntimeError("Database Connection not existent for the current database. Please setup in vault cluster")
+        vault_database_roles = self.vault_client.secrets.database.list_roles(mount_point=mount_point)["data"]["keys"]
+        if role_name not in vault_database_roles:
+            raise RuntimeError("Role name does not exist within the current connection. Please setup the role in vault connection")
+        response = self.vault_client.secrets.database.generate_credentials(name=role_name, mount_point=mount_point)
+        # self.__conf_data(response["data"])
+        if not self.ttl_renew_thread.is_alive():
+            self.ttl_renew_thread.start()
+            self.ttl_renew_thread.join()
+
+    def ttl_expire_check(self) -> None:
```

Comment:
Not entirely sure what value it adds but I believe we want to flag this as an async function 
https://docs.python.org/3/library/asyncio-task.html

Reply:
This is the function that runs in a thread separately from the gestalt's current thread. so want to keep it separate since it is the only code want running on that thread.